docker-compose-4nodes-mt-qlight.yml

```
@@ -0,0 +1,888 @@
+        CONSENSUS_RPC_API="qbft"
+        GENESIS_FILE="/examples/qbft-genesis.json"
+        #set the extradata for 4 nodes
+        UPDATE_QUERY=$${UPDATE_QUERY}' | .extraData = "0xf87aa00000000000000000000000000000000000000000000000000000000000000000f85494d8dba507e85f116b1f7e231ca8525fc9008a6966946571d97f340c8495b661a823f2c2145ca47d63c294e36cbeb565b061217930767886474e3cde903ac594f512a992f3fb749857d758ffda1330e590fa915ec080c0"'
+      fi
+      MPS_GENESIS_FILE=$${DDIR}/mps-genesis.json
+      echo "JQ Update Query:" $${UPDATE_QUERY}
+      jq "$${UPDATE_QUERY}" $${GENESIS_FILE} > $${MPS_GENESIS_FILE}
+      cat $${MPS_GENESIS_FILE}
+      NETWORK_ID=$$(cat $${MPS_GENESIS_FILE} | grep chainId | awk -F " " '{print $$2}' | awk -F "," '{print $$1}')
+      GETH_ARGS_raft="--raft --syncmode full"
+      GETH_ARGS_istanbul="--emitcheckpoints --istanbul.blockperiod 2 --syncmode full"
+      GETH_ARGS_qbft="--emitcheckpoints --istanbul.blockperiod 2 --syncmode full"
+      curl -k -q -X DELETE https://multi-tenancy-oauth2-server:4445/clients/$${PSI}
+      curl -k -s -X POST https://multi-tenancy-oauth2-server:4445/clients \
+        -H "Content-Type: application/json" \
+        --data "{\"grant_types\":[\"client_credentials\"],\"token_endpoint_auth_method\":\"client_secret_post\",\"audience\":[\"Node1\"],\"client_id\":\"$${PSI}\",\"client_secret\":\"foofoo\",\"scope\":\"rpc://eth_* p2p://qlight rpc://admin_* rpc://personal_* rpc://quorumExtension_* rpc://rpc_modules psi://$${PSI}?self.eoa=0x0&node.eoa=0x0\"}" | jq .
+      accessToken=$$(curl -k -s -X POST -F "grant_type=client_credentials" -F "client_id=$${PSI}" -F "client_secret=foofoo" -F "scope=rpc://eth_* p2p://qlight rpc://admin_* rpc://personal_* rpc://quorumExtension_* rpc://rpc_modules psi://$${PSI}?self.eoa=0x0&node.eoa=0x0" -F "audience=Node1" https://multi-tenancy-oauth2-server:4444/oauth2/token | jq '.access_token' -r)
+      echo $${accessToken}
+      geth --datadir $${DDIR} init $${MPS_GENESIS_FILE}
+      geth \
+        --identity node$${NODE_ID}-${QUORUM_CONSENSUS:-istanbul} \
+        --datadir $${DDIR} \
+        --permissioned \
+        --nodiscover \
+        --verbosity 5 \
+        --networkid $${NETWORK_ID} \
+        --rpc \
+        --rpccorsdomain "*" \
+        --rpcvhosts "*" \
+        --rpcaddr 0.0.0.0 \
+        --rpcport 8545 \
+        --rpcapi admin,eth,debug,miner,net,shh,txpool,personal,web3,quorum,quorumExtension,$${CONSENSUS_RPC_API} \
+        --port 21000 \
+        --unlock 0 \
+        --allow-insecure-unlock \
+        --qlight.client \
+        --qlight.client.psi $${PSI} \
+        --qlight.client.token "bearer $${accessToken}" \
+        --qlight.client.serverNode "enode://ac6b1096ca56b9f6d004b779ae3728bf83f8e22453404cc3cef16a3d9b96608bc67c4b30db88e0a5a6c6390213f7acbe1153ff6d23ce57380104288ae19373ef@172.16.239.11:23000?discport=0&raftport=50401" \
+        --qlight.client.serverNodeRPC "https://172.16.239.11:8545/?PSI=$${PSI}" \
```

about the naming, could we make it more fluent?
```suggestion
        --qlight.client.p2p.enr "enode://ac6b1096ca56b9f6d004b779ae3728bf83f8e22453404cc3cef16a3d9b96608bc67c4b30db88e0a5a6c6390213f7acbe1153ff6d23ce57380104288ae19373ef@172.16.239.11:23000?discport=0&raftport=50401" \
        --qlight.client.rpc.url "https://172.16.239.11:8545/?PSI=$${PSI}" \
```

---

Well - I think `client.serverNode*` makes it more obvious that the configured node is a qlight client consuming resources from a server node. The names you suggested do not highlight the roles in the client/server relationship.